smith.lic

```diff
@@ -12,47 +12,75 @@ class Smith
   include DRCT
 
   def initialize(args)
+    ensure_copper_on_hand(5000)
+
     material, item_name, *options = args
+    settings = get_settings(['crafting'])
+    discipline = settings['Blacksmithing']
+    recipes = settings.crafting_recipes.select { |recipe| recipe['type'] =~ /blacksmithing|weaponsmithing|armorsmithing/i }
+    recipe = recipe_lookup(recipes, item_name)
 
-    @settings = get_settings(['crafting'])
+    parts = recipe['part'] || []
+    temper = options.find { |x| x =~ /temper/i }
```

Comment:
It's a question of preference, but you _might_ prefer [Enumberable#grep](http://ruby-doc.org/core-2.0.0/Enumerable.html#method-i-grep) here.

`options.grep(/temper/i).first`